planner/core/rule_partition_processor.go

```diff
@@ -476,21 +476,21 @@ func (s *partitionProcessor) processHashOrKeyPartition(ds *DataSource, pi *model
 // listPartitionPruner uses to prune partition for list partition.
 type listPartitionPruner struct {
 	*partitionProcessor
-	ctx             sessionctx.Context
-	pi              *model.PartitionInfo
-	partitionNames  []model.CIStr
-	colIDToUniqueID map[int64]int64
-	fullRange       map[int]struct{}
-	listPrune       *tables.ForListPruning
+	ctx            sessionctx.Context
+	pi             *model.PartitionInfo
+	partitionNames []model.CIStr
+	fullRange      map[int]struct{}
+	listPrune      *tables.ForListPruning
 }
 
 func newListPartitionPruner(ctx sessionctx.Context, tbl table.Table, partitionNames []model.CIStr,
-	s *partitionProcessor, conds []expression.Expression, pruneList *tables.ForListPruning) *listPartitionPruner {
-	colIDToUniqueID := make(map[int64]int64)
-	for _, cond := range conds {
-		condCols := expression.ExtractColumns(cond)
-		for _, c := range condCols {
-			colIDToUniqueID[c.ID] = c.UniqueID
+	s *partitionProcessor, conds []expression.Expression, pruneList *tables.ForListPruning, columns []*expression.Column) *listPartitionPruner {
+	for i := range pruneList.PruneExprCols {
+		for _, col := range columns {
+			if col.ID == pruneList.PruneExprCols[i].ID {
+				pruneList.PruneExprCols[i].UniqueID = col.UniqueID
```

Comment:
It seems like `pruneList` is from a global object, so if we update it here, will it cause some data race issues?

Reply:
Thank you for spotting it, I will try to clone it.